src/ChargeClustering/ChargeClustering.jl

```diff
@@ -44,28 +51,36 @@ function cluster_detector_hits(
                 esum = ustrip(esum_u)
                 if esum ≈ 0
                     push!(r_pos, mean(c_hits.pos))
+                    push!(r_thit, mean(c_hits.thit))
                 else
                     weights = ustrip.(c_hits.edep) .* inv(esum)
                     push!(r_pos, sum(c_hits.pos .* weights))
+                    push!(r_thit, sum(c_hits.thit .* weights))

```

Comment:
You could ensure to convert `r_pos` back to its original type (`StaticVector` or `CartesianPoint`) when returning this.

Reply:
@oschulz ?